packages/react/src/products/hooks/__tests__/useProductDetails.test.tsx

```diff
@@ -0,0 +1,104 @@
+import { fetchProductDetails } from '@farfetch/blackout-redux/products';
+import {
+  mockGroupedEntries,
+  mockProductId,
+  mockProductsState,
+} from 'tests/__fixtures__/products';
+import { mockStore } from '../../../../tests/helpers';
+import { Provider } from 'react-redux';
+import { renderHook } from '@testing-library/react-hooks';
+import React from 'react';
+import useProductDetails from '../useProductDetails';
+
+jest.mock('@farfetch/blackout-redux/products', () => ({
+  ...jest.requireActual('@farfetch/blackout-redux/products'),
+  fetchProductDetails: jest.fn(() => ({ type: 'get' })),
+}));
+
+// eslint-disable-next-line react/prop-types
+const ReduxProviderWrapper =
+  storeState =>
+  ({ children }) => {
+    const finalStoreState = {
+      ...mockProductsState,
+      ...storeState,
+    };
+
+    return <Provider store={mockStore(finalStoreState)}>{children}</Provider>;
+  };
+
+const renderHookWithRedux = (props, storeState = undefined) => {
+  return renderHook(() => useProductDetails(props), {
+    wrapper: ReduxProviderWrapper(storeState),
+  });
+};
+
+describe('useProductDetails', () => {
+  beforeEach(() => {
+    jest.clearAllMocks();
+  });
+
+  describe('when it has a product id', () => {
+    it('should return the correct values', async () => {
```

Comment:
Shouldn't this be "should return values correctly with initial state", using the previously commented `mockProductDetailsState`, instead of creating a state just for this test? 🤔